src/panels/lovelace/editor/hui-yaml-editor.ts

```diff
@@ -2,15 +2,28 @@ import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
 import { TemplateResult } from "lit-html";
 import "@polymer/paper-input/paper-textarea";
 
+import { HomeAssistant } from "../../../types";
 import { fireEvent } from "../../../common/dom/fire_event";
+import { getCardConfig } from "../common/data";
 
 export class HuiYAMLEditor extends LitElement {
   public yaml?: string;
+  public cardId?: string;
+  private _hass?: HomeAssistant;
 
   static get properties(): PropertyDeclarations {
-    return {
-      yaml: {},
-    };
+    return { yaml: {}, cardId: { type: String } };
+  }
+
+  set hass(value: HomeAssistant) {
+    this._hass = value;
+    if (!this.yaml || this.yaml === "") {
```

Comment:
Don't do it here, check in `updated()` if cardId has changed or not. If it has, fetch yaml.

Reply:
Do you mean `shouldUpdate`?

Reply:
No. Only the lifecycle event `updated` is to be used for side effects. Side effects are things that mutate state, or start things that will mutate state (like fetching data)